File: SecureBlackbox Samples/PDF Encryptor/java/pdfencryptor.java
/*
 * SecureBlackbox 2022 Java Edition - Sample Project
 *
 * This sample project demonstrates the usage of SecureBlackbox in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/secureblackbox
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */

import java.io.*;
import secureblackbox.*;

public class pdfencryptor extends ConsoleDemo {

	public static Certificate loadCertificate(String file, String password)
	{
		Certificate cert = null;

		if (file.length() > 0)
		{
			try
			{
				Certificatemanager certmanager = new Certificatemanager();

				certmanager.importFromFile(file, password);

				cert = certmanager.getCertificate();
			}
			catch (Exception e)
			{
				System.out.println("Cannot load certificate!");
			}
		}

		return cert;
	}

	public static void main(String[] args) {
		if (args.length < 6) {
			System.out.println("usage: pdfencryptor [-input input_file] [-output output_file] [-cert certificate_file] [-certpass certificate_password] [-pass key_password]");
			System.out.println("                    [-encalg encryption_algorithm] [-nometadata]");
			System.out.println("Options: ");
			System.out.println("  -input        An input PDF file to encrypt (Required).");
			System.out.println("  -output       Where the encrypted PDF file will be saved (Required).");
			System.out.println("  -cert         The certificate used to encrypt file.");
			System.out.println("  -certpass     The password for the certificate.");
			System.out.println("  -pass         The password for the encrypting.");
			System.out.println("  -encalg       The encryption algorithm to use. Valid values: RC4, AES128, AES192, AES256, Camellia128, Camellia192, Camellia256, SEED.");
			System.out.println("  -nometadata   Specifies metadata should not be encrypted.");
			System.out.println("\r\nExample: pdfencryptor -input C:\\pdf\\myfile.pdf -output C:\\pdf\\mysignedfile.pdf -cert C:\\certs\\mycert.pfx -certpass mypassword");
		} else {
			try {
				Pdfencryptor encryptor = new Pdfencryptor();
				System.out.println("***************************************************************************************************");
				System.out.println("* This sample illustrates the use of PDFEncryptor component for encrypting PDF documents. *");
				System.out.println("***************************************************************************************************\n");

				String certFile = "";
				String certPass = "";
				String userPass = "";

				for (int i = 0; i < args.length; i++) {
					if (args[i].startsWith("-")) {
						if (args[i].equals("-input"))
							encryptor.setInputFile(args[i + 1]); // args[i+1] corresponds to the value of argument [i]
						if (args[i].equals("-output"))
							encryptor.setOutputFile(args[i + 1]);
						if (args[i].equals("-encalg"))
							encryptor.setEncryptionAlgorithm(args[i + 1]);
						if (args[i].equals("-nometadata"))
							encryptor.setEncryptMetadata(false);
						if (args[i].equals("-cert"))
							certFile = args[i + 1];
						if (args[i].equals("-certpass"))
							certPass = args[i + 1];
						if (args[i].equals("-pass"))
							userPass = args[i + 1];
					}
				}

				if (userPass.isEmpty() && certFile.isEmpty())
				{
					System.out.println("-cert or -pass is required");
					return;
				}

				if (userPass.isEmpty())
				{
					encryptor.setEncryptionType(Pdfencryptor.petCertificate);
					encryptor.setEncryptionCertificate(loadCertificate(certFile, certPass));
				}
				else
				{
					encryptor.setEncryptionType(Pdfencryptor.petPassword);
					encryptor.setUserPassword(userPass);
				}

				encryptor.encrypt();

				System.out.println("PDF file successfully encrypted");
			}
			catch (Exception ex)
			{
				displayError(ex);
			}
		}
	}
}


class ConsoleDemo {
  private static BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

  static String input() {
    try {
      return bf.readLine();
    } catch (IOException ioe) {
      return "";
    }
  }
  static char read() {
    return input().charAt(0);
  }

  static String prompt(String label) {
    return prompt(label, ":");
  }
  static String prompt(String label, String punctuation) {
    System.out.print(label + punctuation + " ");
    return input();
  }

  static String prompt(String label, String punctuation, String defaultVal)
  {
	System.out.print(label + " [" + defaultVal + "] " + punctuation + " ");
	String response = input();
	if(response.equals(""))
		return defaultVal;
	else
		return response;
  }

  static char ask(String label) {
    return ask(label, "?");
  }
  static char ask(String label, String punctuation) {
    return ask(label, punctuation, "(y/n)");
  }
  static char ask(String label, String punctuation, String answers) {
    System.out.print(label + punctuation + " " + answers + " ");
    return Character.toLowerCase(read());
  }

  static void displayError(Exception e) {
    System.out.print("Error");
    if (e instanceof SecureBlackboxException) {
      System.out.print(" (" + ((SecureBlackboxException) e).getCode() + ")");
    }
    System.out.println(": " + e.getMessage());
    e.printStackTrace();
  }
}

File: SecureBlackbox Samples/PDF Encryptor/python/pdf_encryptor.py
# 
# SecureBlackbox 2022 Python Edition - Sample Project
# 
# This sample project demonstrates the usage of SecureBlackbox in a 
# simple, straightforward way. It is not intended to be a complete 
# application. Error handling and other checks are simplified for clarity.
# 
# www.nsoftware.com/secureblackbox
# 
# This code is subject to the terms and conditions specified in the 
# corresponding product license agreement which outlines the authorized 
# usage and restrictions.
# 

import sys
import string
from secureblackbox import *

input = sys.hexversion<0x03000000 and raw_input or input

def ensureArg(args, prompt, index):
  if len(args) <= index:
    while len(args) <= index:
      args.append(None)
    args[index] = input(prompt)
  elif args[index] == None:
    args[index] = input(prompt)

def displayHelp():
    print(
        "NAME\n"
        "  pdfencryptor -- SecureBlackbox PDFEncryptor Demo Application\n\n"
        "SYNOPSIS\n"
        "  pdfencryptor [-input input_file] [-output output_file] [-cert certificate_file] [-certpass certificate_password]\n"
        "             [-pass user_password] [-encalg encryption_algorithm] [-nometadata]n\n"
        "DESCRIPTION\n"
        "  PDFEncryptor demonstrates the usage of PDFEncryptor from SecureBlackbox.\n"
        "  Used to encrypt pdf documents.\n\n"
        "  The options are as follows:\n\n"
        "  -input        An input file to encrypt (Required).\n\n"
        "  -output       Where the encrypted file will be saved (Required).\n\n"
        "  -cert         The certificate used to encrypt file.\n\n"
        "  -certpass     The password for the certificate.\n\n"
        "  -pass         The password for encryption.\n\n"
        "  -encalg       The encryption algorithm to use. Valid values: RC2, RC4, DES, 3DES, AES128, AES192, AES256, Blowfish \n\n"
        "  -nometadata   Specifies metadata should not be encrypted.\n\n"
        "EXAMPLES\n"
        "  pdfencryptor -input C:\\pdf\\helloworld.pdf -output C:\\pdf\\mypdf.scs -cert C:\\certs\\mycert.pfx -certpass mypassword \n\n"
        "  pdfencryptor -input C:\\pdf\\helloworld.pdf -output C:\\pdf\\mypdf.scs -pass mypassword -encalg AES128 -nometadata \n\n"
    )
    
if (len(sys.argv) < 5):
    displayHelp()
    sys.exit(1)
else:
    encryptor = PDFEncryptor()
    cm = CertificateManager()
    
    inputF = ""
    outputF = ""
    certF = ""
    certpass = ""
    password = ""

    for x in range(len(sys.argv)):
        if (sys.argv[x].startswith("-")):
            if (sys.argv[x].lower() == "-input"):
                inputF = sys.argv[x+1]
            if (sys.argv[x].lower() == "-output"):
                outputF = sys.argv[x+1]
            if (sys.argv[x].lower() == "-cert"):
                certF = sys.argv[x+1]
            if (sys.argv[x].lower() == "-certpass"):
                certpass = sys.argv[x+1]
            if (sys.argv[x].lower() == "-pass"):
                password = sys.argv[x+1]
            if (sys.argv[x].lower() == "-encalg"):
                encryptor.set_encryption_algorithm(sys.argv[x+1])
            if (sys.argv[x].lower() == "-nometadata"):
                encryptor.set_encrypt_metadata(False)
    
    if (inputF == ""):
        print("-input is required.\n")
        displayHelp()
        sys.exit(1)
    
    if (outputF == ""):
        print("-output is required.\n")
        displayHelp()
        sys.exit(1)
    
    if (certF != ""):
        cm.import_from_file(certF, certpass)
        encryptor.set_encryption_certificate_handle(cm.get_cert_handle())
    else:
        encryptor.set_user_password(password)

    try:
        encryptor.set_input_file(inputF)
        encryptor.set_output_file(outputF)

        encryptor.encrypt()
    
        print("PDF file successfully encrypted.")
    except Exception as e: 
        print(e)



